packages/core/src/external/fhir/patient/index.ts

```diff
@@ -8,18 +8,70 @@ import {
 import { Address } from "../../../domain/address";
 import { Contact, ContactTypes } from "../../../domain/contact";
 import { driversLicenseURIs, identifierSytemByType } from "../../../domain/oid";
-import { GenderAtBirth, Patient, splitName } from "../../../domain/patient";
+import { GenderAtBirth as MetriportGender, Patient, splitName } from "../../../domain/patient";
 import { getIdFromSubjectId, getIdFromSubjectRef } from "../shared";
+import { IheGender } from "../../carequality/ihe-gateway-v2/outbound/xcpd/process/schema";
+
+type FhirGender = NonNullable<FHIRPatient["gender"]>;
 
 export type PatientIdAndData = Pick<Patient, "id" | "data">;
 
-const genderMapping: { [k in GenderAtBirth]: "female" | "male" } = {
+const iheGenderToFhirGender: Record<IheGender, FhirGender> = {
+  F: "female",
+  M: "male",
+  UN: "other",
+  UNK: "unknown",
+};
+
+const metriportGenderToFhir: Record<MetriportGender, FhirGender> = {
   F: "female",
   M: "male",
+  O: "other",
+  U: "unknown",
+};
+
+const fhirGenderToMetriportGender: Record<FhirGender, MetriportGender> = {
+  female: "F",
+  male: "M",
+  other: "O",
+  unknown: "U",
+};
+
+const fhirGenderToIheGender: Record<FhirGender, IheGender> = {
+  female: "F",
+  male: "M",
+  other: "UN",
+  unknown: "UNK",
```

Comment:
I think those should be each on its own specific folder. Definitely on standard file names so we can easily find them.

I understand the idea to have it all concentrated in one place, but just like we have CW's Patient in a diff folder than CQ's, FHIR's, and Metriport's, these too should be on their respective folder.

The reason is to avoid circular dependency and minimize interdependency in general.
- `iheGenderToFhirGender` and `fhirGenderToIheGender` in CQ's folder
- `metriportGenderToFhir` and `fhirGenderToMetriportGender` in FHIR's

This is assuming:
- Metriport is the most important domain, we should not refer to the ones below;
- FHIR is an addition to it, so it can refer to Metriport's code - but not HIE's;
- HIEs (CQ, CW) are an addition to the first two, so they can refer to FHIR and Metriport.

Reference: [Layered/Onion Architecture](https://www.notion.so/metriport/Development-Best-Practices-bb3574d3c83d46dc9914a9e5c5542ca7?pvs=4#3c5e5579c5f349d9b29ac1d17dc8472d)

Reply:
ditto for the mapping functions below

Reply:
done